3rdparty/find_dependencies.cmake

```diff
@@ -901,11 +901,13 @@ if(BUILD_GUI)
     set(FILAMENT_MATC "${FILAMENT_ROOT}/bin/matc")
     target_link_libraries(3rdparty_filament INTERFACE Threads::Threads ${CMAKE_DL_LIBS})
     if(UNIX AND NOT APPLE)
-        find_library(CPP_LIBRARY c++)
-        if(CPP_LIBRARY)
-            # Ensure that libstdc++ gets linked first
-            target_link_libraries(3rdparty_filament INTERFACE -lstdc++ ${CPP_LIBRARY})
-        endif()
+        find_library(CPPABI_LIBRARY c++abi PATH_SUFFIXES llvm-7/lib llvm-8/lib
```

Comment:
> In theory, we should be using the same version `libc++` and `libc++abi` as the clang compiler. E.g. `clang-7` should go with `llvm-7/lib` and etc.

That will need some careful design to account for all cases (user installed clang, multiple versions installed, downloaded filament, etc.). I think relying on libc++ ABI version compatibility should be OK.

> At least, we can search the llvm libs in the reverse order `llvm-11/lib` -> `llvm-10/lib` ->, etc., which is the same order as we search the clang compiler. (Also add `clang-11` if it is appropriate.)

Added, and reversed order.

> Another issue is that, in my machine (Ubuntu 18.4 + `clang-7` + `libc++-7-dev` + `libc++abi-7-dev`), CMake actually doesn't respect the `PATH_SUFFIXES`, it gets:
> 
> ```shell
> -- CPPABI_LIBRARY: /usr/lib/x86_64-linux-gnu/libc++abi.so                                                                                                
> -- CPP_LIBDIR: /usr/lib/x86_64-linux-gnu                                                                                                                 
> -- CPP_LIBRARY: /usr/lib/x86_64-linux-gnu/libc++.so
> ```
> 
> which, are not under `llvm-7/lib`, but they are symbolic links:
> 
> ```shell
> libc++abi.so -> ../llvm-7/lib/libc++abi.so
> libc++.so -> ../llvm-7/lib/libc++.so
> ```
> 
> so it worked fine but it would be good to force CMake to respect the `PATH_SUFFIXES`.

This ensures that the system default `libc++` is used for the build (so that its the same that will be linked at runtime) There may be multiple `libc++` versions installed on the system; the selected version (eg Ubuntu version specific default, or user chosen with `update-alternatives` is symlinked to /usr/lib/x86_64-linux-gnu/). `PATH_SUFFIXES` are searched if a system default `libc++` is not set.